bow.go

```diff
@@ -45,6 +45,7 @@ type Bow interface {
 	FindFirst(colIndex int, value interface{}) (rowIndex int)
 
 	GetMetadata() Metadata
+	SetMetadata(key, value []string) Bow
```

Comment:
As python parquet and Panda use metadata with one ey only containing json marshalled metadata for there own scope I thin the func should take (key, value string) as argument instead. It's simpler and faster for the only use cases we know about today.